### ISSUE TITLE
Collections cannot be declared in CoffeeScript

### ISSUE THREAD

**Issue by codyhatch** [PaulUithol/Backbone-relational#59]
I have a Backbone app using Backbone relational.  My relations do not work at all if my Collections are declared in CoffeeScript - if I use JavaScript, they work fine.

Models and collections in CoffeeScript:

``` coffeescript
class App.Models.Person extends Backbone.RelationalModel
  urlRoot: '/api/people'
  relations: [{
    type: Backbone.HasOne,
    key: 'worksFor',
    relatedModel: 'App.Models.Company',
    collectionType: 'App.Collections.Companies',
    reverseRelation: {
      key: 'worksAt'
      includeInJSON: 'id'
    }
  }]

class App.Models.Company extends Backbone.RelationalModel
  urlRoot: '/api/companies'

class App.Collections.People extends Backbone.Collections
  model: App.Models.Person
  url: '/api/people'

class App.Collections.Companies extends Backbone.Collections
  model: App.Models.Company
  url: '/api/companies'
```

This does _not_ work.  But if I delete the collections and replace them with their JS equivalents, everything works:

``` javascript
                App.Collections.People = Backbone.Collection.extend({
                    model: App.Models.Person,
                    url: '/api/people/'
                });

                App.Collections.Companies = Backbone.Collection.extend({
                    model: App.Models.Company,
                    url: '/api/companies/'
                });
```

The error given is:

Uncaught TypeError: Cannot read property 'prototype' of undefined

On line [1113](https://github.com/PaulUithol/Backbone-relational/blob/master/backbone-relational.js#L1113) of backbone-relational.js.

Basically, the model attribute doesn't seem to be getting set (it's the "this.model" that throws the error) in the CoffeeScript version - but the rest of backbone works fine, which leads me to think this is a backbone-relational issue.

Can provide more of the code if that helps.  :)
